### DISCUSSION TITLE
Script for updating Zigbee2Mqtt

### DISCUSSION THREAD

**Discussion by jascdk** [tteck/Proxmox#2722] [Ideas]
Hi:)

Would it be possible to make a script for "automating" z2m update?

The commands are like this

# Stop Zigbee2MQTT and go to directory
sudo systemctl stop zigbee2mqtt
cd /opt/zigbee2mqtt

# Backup configuration
cp -R data data-backup

# Update
git pull
npm ci

# Restore configuration
cp -R data-backup/* data
rm -rf data-backup

# Start Zigbee2MQTT
sudo systemctl start zigbee2mqtt

The whole TTECK site is great with a lot of neat scripts - but it is a little tricky to update Z2M - therefor a script would be nice:)

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/blob/2e36b903ada6c7ec0b5185e43edaf1e811d3207f/ct/zigbee2mqtt.sh#L56-L148

**Reply by tteck** [reply]
![Screenshot 2024-04-02 6 04 55 AM](https://github.com/tteck/Proxmox/assets/21342670/797f7b74-65b2-4e48-9bcd-1a04bdc74c0a)

**Comment by jascdk** [comment]
ahhh great - thanks a lot :)

**Comment by chadsell** [comment]
For what it's worth, Zigbee2MQTT have also added their own update script. When I want to update z2m, I just go to /opt/zigbee2mqtt and run ./update.sh.

**Comment by calas80** [comment]
I can't update Z2M in Alpine. I tried running apk update but I'm still at version [1.34.0], but the latest version is 1.37.1. Can anyone help me?

**Reply by chadsell** [reply]
Have you tried what I posted above? Navigate to the folder that Zigbee2MQTT lives in. Then: 
$./update.sh

**Reply by calas80** [reply]
yes bat my /opt (Alpine) is empty 🥲

**Reply by tteck** [reply]
@calas80 it will not update until the package gets updated
https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=v3.19&repo=&arch=x86_64&maintainer=

you can switch to the edge branch, it's a little newer
https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=edge&repo=&arch=x86_64&maintainer=

**Reply by calas80** [reply]
How can I make the Switch? Thank you

**Reply by tteck** [reply]
https://wiki.alpinelinux.org/wiki/Repositories

**Reply by chadsell** [reply]
Ah, I didn't realize z2m was available as a package for Alpine. Good to know! I'm used to working with the manually installed version on Debian.

**Reply by calas80** [reply]
I solved it by switching to the edge version. Thank you
